Add dynamic foveation enable to ovr_performance

NeoSpark314 opened the pull request:
This PR adds a helper function in the ovr_performance gdns class to enable or disable dynamic foveated rendering as described in https://github.com/GodotVR/godot_oculus_mobile/issues/79.
Let me know if sth. should be changed.

Here are some test shots where the foveation level changes:

![dyn_fov_shot01](https://user-images.githubusercontent.com/1610621/71925292-03ae6080-3191-11ea-9817-aedf19a41b6b.jpg)
![dyn_fov_shot02](https://user-images.githubusercontent.com/1610621/71925299-07da7e00-3191-11ea-973a-7702a4742e5d.jpg)
![dyn_fov_shot03](https://user-images.githubusercontent.com/1610621/71925316-0d37c880-3191-11ea-8fbb-164ab996ebec.jpg)

m4gr3d commented on src/config/ovr_performance.cpp:
```diff
@@ -96,6 +99,32 @@ GDCALLINGCONV godot_variant set_foveation_level(godot_object *, void *, void *p_
+			int foveation_available = vrapi_GetSystemPropertyInt(ovr_java, VRAPI_SYS_PROP_FOVEATION_AVAILABLE);
+			if (foveation_available == VRAPI_TRUE) {
+				// Retrieve if set enable or disable
+				bool enable_dynamice_foveation = api->godot_variant_as_bool(p_args[0]);
```
Typo on <pre>enable_dynami<b><i>ce</i></b>_foveation</pre>

m4gr3d commented on src/config/ovr_performance.cpp:
```diff
@@ -96,6 +99,32 @@ GDCALLINGCONV godot_variant set_foveation_level(godot_object *, void *, void *p_
+			if (foveation_available == VRAPI_TRUE) {
+				// Retrieve if set enable or disable
+				bool enable_dynamice_foveation = api->godot_variant_as_bool(p_args[0]);
+				bool dynamic_foveation_valid = false;
```
Should the return value be set to `false` when `dynamic_foveation_valid` is false to indicate that dynamic foveation was not enabled.
Alternatively, instead of returning a `bool`, you could return enums posing at int to indicate:

- FOVEATION_UNAVAILABLE: -2
- DYNAMIC_FOVEATION_UNAVAILABLE: -1
- DYNAMIC_FOVEATION_AVAILABLE: 0 

Just a thought, so let me know what you think.

NeoSpark314 replied:
Right, thanks a lot; this is what I wrote in the documentation of the function in the header... The return value should be dynamic_foveation_valid. (now we would get nil if foveation is completely unavailable, false if dynamic foveation is not available and true if it is.)

Regardin the enum:
If you see common use-cases for this granularity of information on the gdscript API this could be an option; but if we put these kind of detailed return information on all functions we expose it might be a bit much for the average user as this will result in bigger documentation I think (and also a lot of work to make and keep the API consistent).

If someone needs this information a user could also use the ovr_vr_api_proxy to get to the details; but this would require of course much more knowledge about the target devices and the VrApi.

m4gr3d replied:
Thanks for the clarification! 
What you described with using `dynamic_foveation_valid` as the return value instead of `enable_dynamic_foveation`,  should indeed provide enough information, and removes the need for the enum.

m4gr3d commented:
@NeoSpark314 Did you get a chance to apply the discussed changes?

NeoSpark314 commented:
Ah; sorry; I did them immediately and force-pushed but forgot to write something. I fixed the typo and changed to the correct return value in https://github.com/GodotVR/godot_oculus_mobile/commit/5b109bc3418894330be581a95a2f38243d497301

NeoSpark314 commented:
Oh; wait; something went wrong; I will recheck and push and then write again.

NeoSpark314 commented:
Now it should be correct in https://github.com/GodotVR/godot_oculus_mobile/commit/9615dfcc088a832d6d8e3c14f48e73b2f3eec316. Typo fixed and return value as discussed. If you see something else to change let me know.
